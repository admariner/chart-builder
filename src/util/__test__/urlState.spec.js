import Store from '../Store'
import {
  restoreFromStateString,
  getStateString,
  getStateUrl
} from '../urlState'

describe('getStateString', () => {
  it('works', () => {
    const store = Store.create()
    expect(getStateString(store)).toMatchSnapshot()
  })
})

describe('getStateUrl', () => {
  it('works', () => {
    const store = Store.create()
    expect(getStateUrl(store)).toMatchSnapshot()
  })
})

describe('restoreFromStateString', () => {
  it('works', () => {
    const store = Store.create({
      config: {
        mark: 'line'
      }
    })
    const str = getStateString(store)

    const newStore = Store.create({
      config: {
        mark: 'bar'
      }
    })
    restoreFromStateString(newStore, str, {})
    expect(newStore).toMatchSnapshot()
  })
})
