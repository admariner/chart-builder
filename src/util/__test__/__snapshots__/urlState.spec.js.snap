// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStateString works 1`] = `"N4IgbgpgTgzglgewHYgFwEYA0IDGyBmcA5mqAC5xkA2EaSArlVdgLYCGUA1miAEYchsEJHgAmcJERhoA2gF1WbBmyoBlAA4QcdRsxAB3OKLIALHU2wmIxE2XNUAvg6A"`;

exports[`getStateUrl works 1`] = `"http://localhost/?s=N4IgbgpgTgzglgewHYgFwEYA0IDGyBmcA5mqAC5xkA2EaSArlVdgLYCGUA1miAEYchsEJHgAmcJERhoA2gF1WbBmyoBlAA4QcdRsxAB3OKLIALHU2wmIxE2XNUAvg6A"`;

exports[`restoreFromStateString works 1`] = `
Object {
  "config": Object {
    "encodings": Array [],
    "height": null,
    "manualSpec": null,
    "mark": "line",
    "title": null,
    "width": null,
  },
  "fields": Array [],
  "location": undefined,
  "token": "",
}
`;
